master/buildbot/process/buildstep.py

```diff
@@ -511,6 +511,9 @@ def setRenderable(res, attr):
             if self.results != CANCELLED:
                 self.results = EXCEPTION
 
+        self.updateSummary()
```

Comment:
I think the problem here is that we call finishStep too early. We don't even know the final step results which are set in https://github.com/buildbot/buildbot/pull/5952/files#diff-e77ab9c0beb5448f34c444ea388a12754b60ec7ba820cc6a0550989f37f3bde4R534. So I think it would be better to move finishStep to a time when everything is finished, i.e. after the self.updateSummary.stop().